### DISCUSSION TITLE
Error using Query Parameters with list or set

### DISCUSSION THREAD

**Discussion by kevindaffaarr** [fastapi/fastapi#9229] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Query

app = FastAPI()


@app.get("/items/")
async def read_items(q: list = Query(default=[])):
    query_items = {"q": q}
    for item in q:
        print(item)
    return query_items
```

### Description

Run the function by calling await read_items(), then the q still has variable type Query([]) that should be just list or []. It make an error "Query object is not iterable". Yesterday it worked, but right now it's not either in production (Google Cloud Run), or dev (my laptop).

### Operating System

Windows

### Operating System Details

Windows in development, Docker in production

### FastAPI Version

0.92.0

### Python Version

3.11.1

**Comment by yinziyan1206** [comment, marked as answer]
It works fine in my environment. It's weird. Did you change or override some code in `fastapi`

**Reply by kevindaffaarr** [reply]
Nope, I've tried it with fresh environment of python 3.11 with only fastapi (and its dependency) installed. But as a highlight, I run it not with API/Open API Docs, I just call the function like below

![image](https://user-images.githubusercontent.com/40964666/223940349-67967a22-0baf-4271-a9f5-96371b8a4cd2.png)

**Reply by yinziyan1206** [reply]
Oops, you just call the function and it should be wrong. `FastApi` will rewrite the args  and call the function when receving one request. You should use `uvicorn` to start it

**Reply by abhint** [reply]
@kevindaffaarr please visit this site, it will help you.
https://fastapi.tiangolo.com/tutorial/first-steps/

**Comment by kevindaffaarr** [comment]
After I deep dive the error in production, I think I have a wrong error reproducement, the problem is not with the Query but something with the JSON:

```
Traceback (most recent call last): File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 419, in run_asgi result = await app( # type: ignore[func-returns-value]
...
ValueError: Out of range float values are not JSON compliant)
```

But I can't reproduce it anymore because it is already fixed without any change in the code. I'll close this discussion and make a new one if it happen again
